src/main/java/jenkins/plugins/collabnet/steps/PublishWebhookStep.java

```diff
@@ -135,6 +155,24 @@ public FormValidation doCheckStatus(@QueryParameter String status) {
             }
             return FormValidation.error("Invalid TeamForge status value");
         }
+
+        public FormValidation doCheckCredentialsId(
+                @QueryParameter final String credentialsId,
+                @AncestorInPath final Item owner) {
+            // TODO make sure that fallback credentials (i.e. TeamForge) exists if
+            // none is selected
+            return FormValidation.ok();
+        }
+
+        public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item owner) {
+            if (owner == null || !owner.hasPermission(Item.CONFIGURE)) {
+                return new StandardUsernameListBoxModel().withEmptySelection();
+            }
+            String useServerUrl = null;
+            return new StandardUsernameListBoxModel()
+                    .withEmptySelection()
+                    .withAll(PublishEventQStep.PublishEventQStepExecution.lookupCredentials(owner, useServerUrl));
```

Comment:
Please create a single lookupCredentials method. There is one in PublishEventQStepExecution, and you have added another one in PublishWebhookStepExecution.  AFAICT, they are the same, if not, then use the method in PublishWebhookStepExecution here rather than making a reference to PublistEventQStepExecution.

Reply:
ah sure.. forgot to change this one.

Reply:
Done moved it to Helper class.